File: mobile/ios/VaultZero/VaultZero/Crypto/CryptoManager.swift
/**
 * VaultZero iOS — CryptoManager (placeholder)
 *
 * iOS-specific crypto operations:
 * - Store vault key in Keychain with kSecAttrAccessibleWhenUnlockedThisDeviceOnly
 * - AES-256-GCM via CryptoKit
 * - Argon2id via Swift-Argon2 or libsodium-swift
 *
 * SECURITY:
 * - NEVER store vault key in UserDefaults or plaintext files
 * - Keychain item protected by device passcode + biometric
 *
 * Implementation: Day 12-13
 */

// TODO: Day 12-13 — implement CryptoManager

File: mobile/android/app/src/main/kotlin/com/vaultzero/crypto/CryptoManager.kt
/**
 * VaultZero — CryptoManager (placeholder)
 *
 * Android-specific crypto operations:
 * - Store/retrieve vault key from Android Keystore (TEE-backed)
 * - AES-256-GCM encrypt/decrypt using Android KeyStore keys
 * - Argon2id via libsodium-jni or BouncyCastle
 *
 * SECURITY:
 * - Keys never leave the TEE (Trusted Execution Environment)
 * - setUserAuthenticationRequired(true) — requires biometric to use key
 *
 * Implementation: Day 10-11
 */

package com.vaultzero.crypto

// TODO: Day 10-11 — implement CryptoManager with Android Keystore
